src/ComfortPay/Message/TransactionStatusRequest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Omnipay\ComfortPay\Message;
+
+use Money\Currency;
+use Money\Currencies\ISOCurrencies;
+use SoapParam;
+
+class TransactionStatusRequest extends AbstractSoapRequest
+{
+    public function getData()
+    {
+        $this->validate('transactionId');
+
+        $data = parent::getData();
+        $data = array_merge($data, [
+            'transactionId' => $this->getTransactionId(),
+        ]);
+
+        return $data;
+    }
+
+    public function sendData($data)
+    {
+        if ($this->getTestmode())
+        {
+            if (intval($data['cid']) % 2 == 0) {
+                return $this->response = new ChargeResponse($this,
+                    ['transactionId' => $data['transactionId'], 'transactionStatus' => '02', 'transactionApproval' => '123']);
+            }
+            return $this->response = new ChargeResponse($this,
+                ['transactionId' => $data['transactionId'], 'transactionStatus' => '00', 'transactionApproval' => '123']);
+        }
+
+        $request = new \stdClass();
+        $request->transactionId = $data['transactionId'];
+
+        $client = $this->getSoapClient();
+        $response = $client->getTransactionStatus($request);
+
+        return $this->response = new ChargeResponse($this, [
```

Comment:
Didn't you wanted to use `TransactionStatusResponse`? I see you created it, but never used.

Reply:
If it's internally the same type within WSDL, you can probably reuse `ChargeResponse` and remove `TransactionStatusResponse`.

Reply:
You are right. I've unified response object for both methods, based on WSDL - `CardTransactionResponse`